### PR TITLE
Utils pytest

### PR DESCRIPTION
**Pull request by HagerSherif**

### CODE REVIEW

File: test_utils.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    Parameters
+    ----------
+    temp_dir: tuple of py._path.local.LocalPath
+            temporary directory for saving the gif
+    """
+    path=temp_dir[0].join("gif.gif")
+    overlay=Overlay(vol_path,mask_path,str(path))
+    overlay.generate_animation()
+    
+    assert os.path.exists(path)
```
Comment (Muhammad-Al-Barbary) [inline test_utils.py:100]: Checking that the overlay is saved might not be enough. Maybe you can save a reference gif and compare generated ones with it.

File: test_utils.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+    assert len(os.listdir(temp_vol_slices)) == len(os.listdir(temp_mask_slices)) == test_num_of_slices  
+    #correct file type 
+    assert all(file_name.endswith(extension)== True       
+               for file_name in os.listdir(temp_vol_slices))  
+    assert all(file_name.endswith(extension)== True 
+               for file_name in os.listdir(temp_mask_slices)) 
+    #correct slice dimenisons 
+    assert (nib.load(temp_vol_slices/file_name).get_fdata().shape==test_shape 
+            for file_name in os.listdir(temp_vol_slices))   
+    assert (nib.load(temp_mask_slices/file_name).get_fdata().shape==test_shape
```
Comment (Muhammad-Al-Barbary) [inline test_utils.py:138]: You will also need to check for the pixels values. Flipping and/or Scaling are two common bugs in slicing, for example.

**Review by Muhammad-Al-Barbary — COMMENTED** [review]
Well done!
Remember to apply the styling convention.

File: test_utils.py [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+
+
+# test input files
+volume_dir="test/testdata/volume"
+mask_dir="test/testdata/mask"
+vol_path=os.path.join(volume_dir,'resized_liver.nii')
+mask_path= os.path.join(mask_dir,'resized_mask.nii')   # 0 background, 1 liver, 2 lesion
+mask=nib.load(mask_path)
+volume=nib.load(vol_path)
```
Comment (helrewaidy) [inline test_utils.py:16]: It'd be great if you can add these to a fixture as well.

**Review by helrewaidy — APPROVED** [review]
Impressive work ya Hager! Thanks

File: test_utils.py (hunk 1 above)
Comment (HagerSherif) [reply test_utils.py]: done

File: test_utils.py (hunk 2 above)
Comment (HagerSherif) [reply test_utils.py]: done

File: test_utils.py (hunk 3 above)
Comment (HagerSherif) [reply test_utils.py]: I put each one in a separate fixture, is that ok?

File: tests/test_utils.py [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+)
+import nibabel as nib
+import numpy as np
+from PIL import Image
+import torch
+import natsort
+
+
+@pytest.fixture
+def volume_dir():
```
Comment (Muhammad-Al-Barbary) [inline tests/test_utils.py:18]: maybe you can add these paths to configs to read them directly instead of creating a fixture for each of them.

**Review by Muhammad-Al-Barbary — APPROVED** [review]
Well done, please make sure all docstrings are up to date before merging.

**Pull request merged**